src/playlist.js

```diff
@@ -0,0 +1,424 @@
+import videojs from 'video.js';
+import { isIndexInBounds, randomize } from './utils.js';
+
+/**
+ * This is standalone class that encapsulates all playlist logic that does not require a `player` instance
+ */
+export default class Playlist extends videojs.EventTarget {
+  /**
+   * Creates a new Playlist instance from a given array of items.
+   *
+   * @param {Object[]} items - An array of objects to initialize the playlist.
+   * @return {Playlist} A new Playlist instance populated with the given items.
+   */
+  static from(items) {
+    const playlist = new Playlist();
+
+    playlist.set(items);
+
+    return playlist;
+  }
+
+  constructor() {
+    super();
+
+    this.list_ = [];
+    this.currentIndex_ = null;
+    this.repeat_ = false;
+    this.log_ = console;
```

Comment:
I think we should remove this.

I see 2 possible solutions here:

1:
Consider logger to be injected via Dependency Injection and fallback to `noop`

2:
consider remove log and expect onError and onWarn callback to provided (as usual `noop` by default).

Reply:
Done